plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -256,25 +257,25 @@ public void resolveDependencies(Plugin plugin) {
 
         for (Plugin dependency : dependentPlugins) {
             String dependencyName = dependency.getName();
-            String dependencyVersion = dependency.getVersion();
+            VersionNumber dependencyVersion = dependency.getVersion();
             if (dependency.getPluginOptional()) {
                 System.out.println("Skipping optional dependency " + dependencyName);
                 continue;
             }
 
-            String installedVersion = "";
+            VersionNumber installedVersion = null;
             if (installedPluginVersions.containsKey(plugin.getName())) {
                 installedVersion = installedPluginVersions.get(dependencyName);
             } else if (bundledPluginVersions.containsKey(plugin.getName())) {
                 installedVersion = bundledPluginVersions.get(dependencyName);
             }
 
-            if (!StringUtils.isEmpty(installedVersion)) {
-                if (comparePluginVersions(installedVersion, dependencyVersion) < 0) {
+            if (installedVersion != null) {
+                if (installedVersion.compareTo(dependencyVersion) < 0 ) {
```

Comment:
Space after the 0, just to stay consistent.
```suggestion
                if (installedVersion.compareTo(dependencyVersion) < 0) {
```

Reply:
Fixed in ba2291f.